protocols/relay/src/behaviour.rs

```diff
@@ -54,6 +54,8 @@ pub struct Config {
     pub reservation_duration: Duration,
     pub reservation_rate_limiters: Vec<Box<dyn rate_limiter::RateLimiter>>,
 
+    pub substream_timeout: Duration,
```

Comment:
```suggestion
    pub stream_timeout: Duration,
```

We are moving away from the **sub**stream term, and instead use **stream** everywhere. See also https://github.com/libp2p/rust-libp2p/pull/3912.